src/platform/tls_openssl.c

```diff
@@ -411,233 +298,812 @@ CxPlatTlsNegotiatedCiphers(
+// and @p bytes_read. If the record is incomplete, it signals OpenSSL to
+// wait for more data.
+//
+// @param[in]  s            Pointer to the SSL connection object.
+// @param[out] buf          Pointer to the buffer containing the record data.
+//                          If no data is available, set to NULL.
+// @param[out] bytes_read   Length of the record returned in @p buf.
+//                          If no data is available, set to 0.
+// @param[in]  arg          Unused argument (typically NULL).
+//
+// @return Always returns 1.
+//
+static int QuicTlsRcvRec(SSL *s, const unsigned char **Buf, size_t *BytesRead,
+                            void *Arg)
+{
+    RECORD_ENTRY *entry;
+    struct AUX_DATA *AData = GetSslAuxData(s);
+    CXPLAT_LIST_ENTRY* lentry;
+
+    UNREFERENCED_PARAMETER(Arg);
+
+    AData = GetSslAuxData(s);
+    CXPLAT_DBG_ASSERT(AData != NULL);
+
+    //
+    // Iterate over our received record list looking
+    // for a complete entry to submit to the TLS
+    // stack
+    //
+    lentry = AData->RList.Flink;
+    while (lentry != &AData->RList) {
+        entry = CXPLAT_CONTAINING_RECORD(lentry, RECORD_ENTRY, Entries);
+        lentry = lentry->Flink;
+        if (entry->Incomplete)
+            return 1;
+        if (entry->FreeMe == 1)
+            continue;
+        *Buf = entry->Record;
+        *BytesRead = entry->RecLen;
+        entry->FreeMe = 1;
+        break;
+  }
+  return 1;
+
+}
+
+//
+// @brief Callback to release a previously buffered TLS record.
+//
+// This function is called by OpenSSL after a TLS record has been fully
+// processed and can be safely released. It verifies the number of bytes
+// read matches the expected record length, frees the associated memory,
+// and resets the pointer.
+//
+// @param[in] bytes_read  The number of bytes processed in the TLS record.
+// @param[in] arg         Unused argument (typically NULL).
+//
+// @return Always returns 1.
+//
+static int QuicTlsRlsRec(SSL *S, size_t BytesRead,
+                            void *Arg)
+{
+    struct AUX_DATA *AData = GetSslAuxData(S);
+    RECORD_ENTRY *entry;
+    CXPLAT_LIST_ENTRY* lentry;
+
+    UNREFERENCED_PARAMETER(Arg);
+
+    //
+    // Look for Entries that are marked for freeing
+    // if the record length matches, we can free it
+    //
+    lentry = AData->RList.Flink;
+    while (lentry != &AData->RList) {
+      entry = CXPLAT_CONTAINING_RECORD(lentry, RECORD_ENTRY, Entries);
+      lentry = lentry->Flink;
+      if ((entry->FreeMe == 1) && (entry->RecLen == BytesRead)) {
+        CxPlatListEntryRemove(&entry->Entries);
+        free(entry->Record);
+        free(entry);
+        return 1;
+      }
+    }
+    return 1;
+
+}
+
+//
+// @brief Callback to yield TLS secrets to the QUIC stack.
+//
+// This function is invoked by OpenSSL to provide traffic secrets during
+// the QUIC handshake. It installs the given secret into the ngtcp2 QUIC
```

Comment:
Wrong QUIC stack😄

Reply:
crud, I converted alot of this from my ngtcp2 efforts, apologies, will fix